driver-core/src/main/java/com/datastax/driver/core/PoolingOptions.java

```diff
@@ -68,191 +83,237 @@
+    private volatile ProtocolVersion protocolVersion;
 
-    private final int[] maxSimultaneousRequestsPerConnection = new int[]{ DEFAULT_MAX_REQUESTS_PER_CONNECTION, DEFAULT_MAX_REQUESTS_PER_CONNECTION, 0 };
+    // The defaults for these fields depend on the protocol version, which is only known after control connection initialization.
+    // Yet if the user set them before initialization, we want to keep their values. So we use -1 to mean "uninitialized".
+    private final int[] coreConnections = new int[]{ UNSET, UNSET, 0 };
+    private final int[] maxConnections = new int[]{ UNSET, UNSET, 0 };
+    private final int[] newConnectionThreshold = new int[]{ UNSET, UNSET, 0 };
+    private volatile int maxRequestsPerConnectionLocal = UNSET;
+    private volatile int maxRequestsPerConnectionRemote = UNSET;
 
-    private final int[] coreConnections = new int[] { DEFAULT_CORE_POOL_LOCAL, DEFAULT_CORE_POOL_REMOTE, 0 };
-    private final int[] maxConnections = new int[] { DEFAULT_MAX_POOL_LOCAL , DEFAULT_MAX_POOL_REMOTE, 0 };
-
-    private volatile int maxSimultaneousRequestsPerHostLocal = DEFAULT_MAX_REQUESTS_PER_HOST_LOCAL;
-    private volatile int maxSimultaneousRequestsPerHostRemote = DEFAULT_MAX_REQUESTS_PER_HOST_REMOTE;
-    
     private volatile int idleTimeoutSeconds = DEFAULT_IDLE_TIMEOUT_SECONDS;
     private volatile int poolTimeoutMillis = DEFAULT_POOL_TIMEOUT_MILLIS;
     private volatile int heartbeatIntervalSeconds = DEFAULT_HEARTBEAT_INTERVAL_SECONDS;
 
     private volatile Executor initializationExecutor = DEFAULT_INITIALIZATION_EXECUTOR;
 
-    public PoolingOptions() {}
+    public PoolingOptions() {
+    }
 
     void register(Cluster.Manager manager) {
         this.manager = manager;
     }
 
     /**
-     * Returns the number of simultaneous requests on a connection below which
-     * connections in excess are reclaimed.
-     * <p>
-     * This option is only used with {@code ProtocolVersion#V2} or below.
-     * <p>
-     * If an opened connection to an host at distance {@code distance}
-     * handles less than this number of simultaneous requests and there is
-     * more than {@link #getCoreConnectionsPerHost} connections open to this
-     * host, the connection is closed.
-     * <p>
-     * The default value for this option is 25 for {@code LOCAL} and
-     * {@code REMOTE} hosts.
+     * Returns the core number of connections per host.
      *
      * @param distance the {@code HostDistance} for which to return this threshold.
-     * @return the configured threshold, or the default one if none have been set.
-     *
-     * @deprecated this option isn't used anymore with the current pool resizing algorithm.
+     * @return the core number of connections per host at distance {@code distance}.
      */
-    @Deprecated
-    public int getMinSimultaneousRequestsPerConnectionThreshold(HostDistance distance) {
-        return 0;
+    public int getCoreConnectionsPerHost(HostDistance distance) {
```

Comment:
Isn't there a slight difference in API in that previously if setCoreConnections(distance) is unused that it'd return the default?  Now it will return Integer.MIN_VALUE?